### PR TITLE
feat: support python code in components

### PR DESCRIPTION
**Pull request by chuang8511**
Because

- we want to use python lib in pdf component

This commit

- add python virtual machine in container

### CODE REVIEW

**Comment by chuang8511** [comment]
@donch1989 
I am curious about if we still use Dockerfile in pipeline-backend.
I searched in instill-core and pipeline-back, but I did not find where we execute it.

**Comment by jvallesm** [comment]
> @donch1989 I am curious about if we still use Dockerfile in pipeline-backend. I searched in instill-core and pipeline-back, but I did not find where we execute it.

Without looking into much detail, I think the step they're used in is the image build GitHub Action.

**Comment by chuang8511** [comment]
@jvallesm 
Thanks for your response! 
I think I understand and found the code! Thank you!!

**Comment by chuang8511** [comment]
@jvallesm @donch1989 
Usually, when we modify Dockerfile rather Dockerfile.dev, how do you test to verify it?
I build the image by Dockerfile, but it seems the container will not be out of instill-core.
I am not quite sure about this part. Do you have some special skills to test it?

**Comment by jvallesm** [comment]
> @jvallesm @donch1989 Usually, when we modify Dockerfile rather Dockerfile.dev, how do you test to verify it? I build the image by Dockerfile, but it seems the container will not be out of instill-core. I am not quite sure about this part. Do you have some special skills to test it?

I haven't faced this use case yet but I'd follow the regular dev env setup replacing `Dockerfile.dev` with `Dockerfile` in the `make build` step.

**Review by jvallesm — DISMISSED** [review]
The changes look good.

**Comment by chuang8511** [comment]
> I haven't faced this use case yet but I'd follow the regular dev env setup replacing Dockerfile.dev with Dockerfile in the make build step.

Yes, I did that before. 
But, it seems that I cannot isolate the container from instill-core & pipeline-backend.
It seems there is a small difference between Dockerfile & Dockerfile.dev, which makes me unable to do it.

**Comment by jvallesm** [comment]
> > I haven't faced this use case yet but I'd follow the regular dev env setup replacing Dockerfile.dev with Dockerfile in the make build step.
> 
> Yes, I did that before. But, it seems that I cannot isolate the container from instill-core & pipeline-backend. It seems there is a small difference between Dockerfile & Dockerfile.dev, which makes me unable to do it.

I see, please address this question on Slack #dev-vdp or Discord as others might have experienced this before.

**Comment by chuang8511** [comment]
Tested with Dockerfile, and it is ok now!

File: Dockerfile.dev [hunk 1]
```diff
@@ -9,6 +9,14 @@ WORKDIR /${SERVICE_NAME}
 
 ARG TARGETOS TARGETARCH K6_VERSION
 
+# Install Python, create virtual environment, and install pdfplumber
+RUN apt update && \
+    apt install -y python3 python3-venv && \
+    python3 -m venv /opt/venv && \
+    /opt/venv/bin/pip install pdfplumber && \
+    apt install -y poppler-utils wv unrtf tidy tesseract-ocr libtesseract-dev && \
```
Comment (donch1989) [inline Dockerfile.dev:17]: We can merge this to L14

**Pull request merged**